executor/pkg/executor/cl_executor.go

```diff
@@ -297,3 +281,67 @@ func (cle *ChainlinkExecutor) Validate() error {
 	}
 	return nil
 }
+
+// GetMessageStatus checks if a message should be executed and/or retried.
+// Returns (shouldRetry bool, shouldExecute bool, error) to indicate whether the message should be retried (added back to heap) and executed.
+func (cle *ChainlinkExecutor) GetMessageStatus(ctx context.Context, message protocol.Message, currentTime int64) (bool, bool, error) {
```

Comment:
Can use named returns here as well to make it clear what each boolean means


```suggestion
func (cle *ChainlinkExecutor) GetMessageStatus(ctx context.Context, message protocol.Message, currentTime int64) (shouldRetry bool, shouldExecute bool, error) {
```

Or a `MessageStatus` struct to be more verbose (not a bad thing here)